### PR TITLE
feat: 🎸 update worker filter field to use code editor

### PR DESCRIPTION
**Pull request by cameronperera**
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-15737

# Description

update worker filter field to use code editor

## Screenshots (if appropriate)
Before:

https://github.com/user-attachments/assets/4916d5a2-b068-44f4-9c57-b412c15ea9f2

After:

https://github.com/user-attachments/assets/c6f4d791-b639-4aa5-9821-7f1ae7955e6d

## Checklist

- [x] I have added before and after screenshots for UI changes
- [ ] I have added JSON response output for API changes
- [ ] I have added steps to reproduce and test for bug fixes in the description
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: ui/admin/tests/acceptance/storage-buckets/create-test.js [hunk 1]
```diff
@@ -58,6 +57,11 @@ module('Acceptance | storage-buckets | create', function (hooks) {
     await fillIn(commonSelectors.FIELD_NAME, commonSelectors.FIELD_NAME_VALUE);
     await select(selectors.FIELD_SCOPE, 'global');
 
+    await fillIn(
```
Comment (lisbet-alvarez) [inline ui/admin/tests/acceptance/storage-buckets/create-test.js:60]: can we also validate that after save is clicked, the codeblock is shown (not the editor)? Or will the read type tests be added in a different PR?

**Review by lisbet-alvarez — APPROVED** [review]
Looks awesome!! 👏

File: ui/admin/tests/acceptance/storage-buckets/create-test.js (hunk 1 above)
Comment (cameronperera) [reply ui/admin/tests/acceptance/storage-buckets/create-test.js:60]: Sorry, I should have called that out in the description. I was going to go back through the tests and add those in afterwards since the focus of this PR is "create".

File: ui/admin/app/components/form/storage-bucket/aws/index.hbs [hunk 2]
```diff
@@ -453,30 +453,33 @@
...
     </F.HelperText>
-    {{#if @model.errors.worker_filter}}
-      <F.Error data-test-worker-filter-error as |E|>
-        {{#each @model.errors.worker_filter as |error|}}
-          <E.Message>{{error.message}}</E.Message>
-        {{/each}}
-      </F.Error>
-    {{/if}}
-  </Hds::Form::TextInput::Field>
+    <F.Control>
```
Comment (ZedLi) [inline ui/admin/app/components/form/storage-bucket/aws/index.hbs:470]: Is this accessible? I don't see a label or an aria-label now that you switched over to a fieldset and only a legend. Should it be `aria-labelledby` to the legend?
